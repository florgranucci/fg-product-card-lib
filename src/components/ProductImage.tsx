import React from 'react';
import { CSSProperties, useContext } from 'react';
import { ProductContext } from './ProductCard';
import noImg from '../assets/no-image.jpg'
import styles from '../styles/styles.module.css'

export interface Props {
    img?: string;
    className?: string;
    style?: CSSProperties;
}
export const ProductImage = ({ img = '', className, style }: Props) => {
    const { product: { img: productImage } } = useContext(ProductContext);
    return(
    <img style={style} className={`${styles.productImg} ${className}`} src={img ? img : productImage || noImg} alt='product'></img>
)}