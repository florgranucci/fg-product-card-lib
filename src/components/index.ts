import { ProductCardHOCProps } from "../interfaces/interfaces";
import { ProductButtons } from "./ProductButton";
import { ProductCard as ProductCardHoc } from "./ProductCard";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";

export { ProductButtons } from "./ProductButton";
export { ProductImage } from "./ProductImage";
export { ProductTitle } from "./ProductTitle";

export const ProductCard:ProductCardHOCProps = Object.assign(ProductCardHoc, {
    Buttons: ProductButtons,
    Image: ProductImage,
    Title: ProductTitle,
})

export default ProductCard