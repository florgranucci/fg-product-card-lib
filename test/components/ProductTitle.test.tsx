import renderer  from "react-test-renderer";
import React from "react";
import { ProductTitle } from '../../src/components/ProductTitle';
import { product1 } from "../data/products";
import { ProductCard } from "../../src/components/ProductCard";

describe('ProductTitle', () => {
  it('renders correctly with custom title', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom title" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly the product title', () => {
    const wrapper = renderer.create(
        <ProductCard product={product1}>
            {
                () => <ProductTitle title={product1.title} />
            }
        </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
