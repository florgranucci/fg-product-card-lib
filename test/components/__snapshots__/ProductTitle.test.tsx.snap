// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle renders correctly the product title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Product 1 sin imagen
  </span>
</div>
`;

exports[`ProductTitle renders correctly with custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom title
</span>
`;
